plugin-server/src/worker/plugins/run.ts

```diff
@@ -212,7 +213,9 @@ async function runSingleTeamPluginComposeWebhook(
 
 export async function runComposeWebhook(hub: Hub, event: PostIngestionEvent): Promise<void> {
     // Runs composeWebhook for all plugins for this team in parallel
-    const pluginMethodsToRun = await getPluginMethodsForTeam(hub, event.teamId, 'composeWebhook')
+    let pluginMethodsToRun = await getPluginMethodsForTeam(hub, event.teamId, 'composeWebhook')
+
+    pluginMethodsToRun = await filterPluginMethodsForActionMatches(hub, event, pluginMethodsToRun)
```

Comment:
(feel like I mentioned this already): It would be better if this was done in parallel to executing the `composeWebhook` plugins, so you can just move the filtering code into `runSingleTeamPluginComposeWebhook` I'd say easier to read also.